src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/PropertySymbolReferenceFinder.cs

```diff
@@ -83,9 +87,28 @@ private static bool IsForEachProperty(IPropertySymbol symbol)
             IPropertySymbol symbol,
             Document document,
             SemanticModel semanticModel,
+            FindReferencesSearchOptions options,
             CancellationToken cancellationToken)
         {
-            var nameReferences = await FindReferencesInDocumentUsingSymbolNameAsync(symbol, document, semanticModel, cancellationToken).ConfigureAwait(false);
+            var nameReferences = await FindReferencesInDocumentUsingSymbolNameAsync(
+                symbol, document, semanticModel, cancellationToken).ConfigureAwait(false); 
+
+            if (options.AssociatePropertyReferencesWithSpecificAccessor)
+            {
+                // We want to associate property references to a specific accessor (if an accessor
+                // is being referenced).  Check if this reference would match an accessor. if so, do
+                // not add it.  It will be added by PropertyAccessorSymbolReferenceFinder.
+                var semanticFacts = document.GetLanguageService<ISemanticFactsService>();
+
+                nameReferences = nameReferences.WhereAsArray(loc =>
```

Comment:
>WhereAsArray [](start = 48, length = 12)

Can this code be shared between the accessor and the property reference finders?

Reply:
They're somewhat reversed.  One is saying ```accessors.Length == 0```, meaning "this didn't bind to a specific accessor", the other is ```accessors.Contains(symbol)```, meaning "this did bind to a specific accessor".

The part we can (and do) share is the GetReferencedAccessorSymbols part.